pymongo/common.py

```diff
@@ -871,6 +870,37 @@ def _ecoc_coll_name(encrypted_fields: Mapping[str, Any], name: str) -> Any:
 WRITE_CONCERN_OPTIONS = frozenset(["w", "wtimeout", "wtimeoutms", "fsync", "j", "journal"])
 
 
+def _get_timeout_details(client: MongoClient) -> dict[str, Any]:
+    details = {}
+    timeout = client.options.timeout
+    socket_timeout = client.options.pool_options.socket_timeout
+    connect_timeout = client.options.pool_options.connect_timeout
+    topology = client.topology_description.server_descriptions()
+    if timeout:
+        details["timeout"] = timeout * 1000
```

Comment:
Would it make more sense to add _csot.get_timeout() here? That would reflect the last CSOT timeout that was set.